dist/restore/index.js

```diff
@@ -0,0 +1,143 @@
+// GCS Cache Main Action - Restore caches
+
+// *** Imports ***
+// Node.js dependencies
+const fs = require('fs');
+const path = require('path');
+const md5 = require('md5');
+// GitHub Actions dependencies
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+// Google Cloud client library
+// https://github.com/googleapis/nodejs-storage
+const {Storage} = require('@google-cloud/storage');
+
+// *** Configuration ***
+const downloadDirectory = 'temp';
+const compressionCommand = 'tar -zcf';
+const expandCommand = 'tar -zxf';
+const compressedFileExtension = '.tar.gz';
+const removeCommand = 'rm -r';
+
+async function run() {
+  try {
+    // Read environment variables
+    const gcsKeyfileString = process.env.IOS_GCS_KEYFILE_BASE64;
+
+    // Read inputs
+    const noDownload = core.getInput('no-download', { required: false });
+    const cacheKey = core.getInput('cache-key', { required: true });
+    const pathsString = core.getInput('paths', { required: true });
+
+    // Extract GCS Keyfile
+    const buffer = new Buffer.from(gcsKeyfileString, 'base64');
+    const gcsKeyfile = JSON.parse(buffer.toString('ascii'));
+
+    // Create a GCS client
+    const storage = await new Storage({ credentials: gcsKeyfile });
+    const bucketName = gcsKeyfile.project_id;
+    const bucket = await storage.bucket(bucketName);
+
+    // Get all caches available in Bucket for the given cache key
+    const [availableCaches] = await bucket.getFiles({
+      prefix: cacheKey
+    });
+
+    if (typeof availableCaches !== undefined && availableCaches.length > 0) {
+      console.log(`Caches for key ${cacheKey} found in bucket ${bucketName}`);
+    } else {
+      console.log(`No caches found for key ${cacheKey} in the bucket ${bucketName}`);
+    }
+
+    // List all given path with their corresponding cache references (if exists in the bucket)
+    const cacheObjects = pathsString.split('\n').map(cachePath => {
+      const cacheFileName = fileNameForCache(cacheKey, cachePath);
+      const cacheReference = availableCaches.find(cache => {
```

Comment:
use proper bracket notation.

```suggestion
      const cacheReference = availableCaches.find((cache) => {
```

Reply:
Done!